hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -562,6 +578,26 @@ private List<BalanceChange> constructBalanceChanges(final CryptoTransferWrapper
                 }
             }
 
+            for (var hbarTransfer : transferOp.transferWrapper().hbarTransfers()) {
+                if (hbarTransfer.sender() != null) {
+                    changes.add(
+                            BalanceChange.changingHbar(
+                                    aaWith(
+                                            hbarTransfer.sender(),
+                                            -hbarTransfer.amount(),
+                                            hbarTransfer.isApproval()),
+                                    EntityIdUtils.accountIdFromEvmAddress(senderAddress)));
```

Comment:
Maybe extract `EntityIdUtils.accountIdFromEvmAddress(senderAddress)` to a variable before all the loops?